### ISSUE TITLE
Bug: MarkdownFileInfo.editor field access throws during initialization

### ISSUE THREAD

**Issue by pjeby** [obsidianmd/obsidian-api#156]
The [Outliner plugin](https://github.com/vslinko/obsidian-outliner/issues/507) gets an error when accessing the MarkdownFileInfo.editor field during editor initialization, like so:

```ts
const { editor } = state.field(obsidian.editorInfoField);
```

This is because Obsidian doesn't guard against there not being an editMode set yet:

```js
            Object.defineProperty(t.prototype, "editor", {
                get: function() {
                    return this.editMode.editor   // <-- this should be `?.editor` 
                },
                enumerable: !1,
                configurable: !0
            }),
```

This issue was apparently introduced in 1.5.0, as I [reported via Discord back in November](https://discord.com/channels/686053708261228577/989603365606531104/1169779476243152967), though at the time I didn't know it occurred on creation of *every* new editor.  Recently it was reported by someone else as an issue I could repro without restarting Obsidian, so I was able to track it down and identify the specific issue.

**Comment by lishid** [comment]
I think it might be better for us to just not initialize the editor with plugins, and then reconfiguring the editor with plugins after - accessing properties that are not ready is always a bad thing to do. I could make this patch but it would still cause plugins to crash when attempting to use editor without checking for null - rather probably easier to ensure it always exists.

**Comment by pjeby** [comment]
Sure.  In Outliner's case it's actually *checking* for a null on the next line, but the Obsidian API throws an error there since it's not initialized.  (I've suggested they work around for now by accessing editMode?.editor directly or try/catch, but I'm not sure how much the author is around -- looks like their last activity predates even 1.5.0 alpha.)

Outliner in fact sits in a setTimeout loop waiting for the editor field to be initialized, so your approach would allow them to drop that in a future release as well.  (I'm assuming you mean that MarkdownFileInfo.editor would lose the optionality, or at least that the value supplied in the state field would be always have a valid .editor at extension init.)

**Comment by lishid** [comment]
> Sure. In Outliner's case it's actually checking for a null on the next line, but the Obsidian API throws an error there since it's not initialized.

Yeah I understand; I think another plugin ran into the same bug as well.

**Comment by lishid** [comment]
I will send you a build to test the fix in discord.

**Comment by pjeby** [comment]
Closing since this appears to be fixed in 1.5.9 beta and 1.5.10 alpha.
